tools/tests/test_backend_datadog.py

```diff
@@ -0,0 +1,178 @@
+            }
+        }
+        query = self.generate_query(rule, config={"tags": ["test-tag"]})
+        expected_query = "@attribute:test AND test-tag:mytag"
+        self.assertEqual(query, expected_query)
+
+    def test_special_characters_escape(self):
+        rule = {
+            "detection": {
+                "selection": {
+                    "attribute": "test",
+                    "regex-attribute": "anything!inbetween",
+                },
+                "condition": "selection",
+            }
+        }
+        query = self.generate_query(rule)
+        expected_query = """@attribute:test AND @regex-attribute:anything\\!inbetween"""
+        self.assertEqual(query, expected_query)
+
+    def test_space_escape(self):
+        rule = {
+            "detection": {
+                "selection": {
+                    "attribute": "test",
+                    "space-attribute": "with space",
+                },
+                "condition": "selection",
+            }
+        }
+        query = self.generate_query(rule)
+        expected_query = "@attribute:test AND @space-attribute:with?space"
+        self.assertEqual(query, expected_query)
+
+    def test_field_mappings(self):
+        query = self.generate_query(
+            self.generate_basic_rule(),
+            fieldmappings={"attribute": FieldMapping("attribute", "another_attribute")},
+        )
+        expected_query = "@another_attribute:test"
+        self.assertEqual(query, expected_query)
+
+    def test_list_selection(self):
+        rule = {
+            "detection": {
+                "selection": {
+                    "attribute": "test",
+                    "list-attribute": [
+                        "a",
+                        "b",
+                    ],
+                },
+                "condition": "selection",
+            }
+        }
+        query = self.generate_query(rule)
+        expected_query = "@attribute:test AND @list-attribute:(a OR b)"
+        self.assertEqual(query, expected_query)
+
+
+if __name__ == "__main__":
+    """
+    Run the Datadog backend over all the Sigma rules in the repository.
+    You can run the script simply with: `python3 -m tests.test_backend_datadog`
+    from the `tools/` directory.
+    """
+    parser = argparse.ArgumentParser(description="Test the Datadog backend over all the Sigma rules in the repository.")
+    parser.add_argument("--verbose", default=False, action="store_true", help="Print individual results about each processed rule.")
+
+    args = parser.parse_args()
+
+    verbose_report = args.verbose
+
+    skipped = 0
+    errors = 0
+    successes = 0
+    total = 0
+
+    config = SigmaConfiguration()
+    backend = DatadogLogsBackend(config)
+
+    for (dirpath, _, filenames) in os.walk("../rules"):
+        for filename in filenames:
+            if filename.endswith(".yaml") or filename.endswith(".yml"):
+                rule_path = os.path.join(dirpath, filename)
```

Comment:
nit: you can remove a couple of line and indentation levels with
```suggestion
    from glob import glob
    for rule_path in glob('../rules/**/*.y*ml', recursive=True):
```